.github/workflows/deploy.yml

```diff
@@ -4,241 +4,42 @@ on:
-            - name: Parse issue info
-              id: parse_issue_info
-              uses: gr2m/get-json-paths-action@v1.x
-              with:
-                  json: ${{ steps.get_issue_info.outputs.data }}
-                  pull_request_url: 'pull_request.url'
-            - name: Fetch PR info
-              uses: octokit/request-action@v2.x
-              id: get_pr_ref
-              with:
-                  route: GET ${{ steps.parse_issue_info.outputs.pull_request_url }}
-              env:
-                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-            - name: Parse PR info
-              id: parse_pr_info
-              uses: gr2m/get-json-paths-action@v1.x
-              with:
-                  json: ${{ steps.get_pr_ref.outputs.data }}
-                  ref: 'head.ref'
-                  sha: 'head.sha'
-            - run: 'echo ${{ steps.parse_pr_info.outputs }}'
-            - uses: actions/checkout@v2
-              with:
-                  token: ${{ secrets.PAT }}
-                  repository: 0xProject/0x-main-infra
-            - name: Update API
-              run: sed -i -e 's/"883408475785\.dkr\.ecr\.us-east-1\.amazonaws\.com\/0x\/api\:.*"/"883408475785\.dkr\.ecr\.us-east-1\.amazonaws\.com\/0x\/api\:${{ steps.parse_pr_info.outputs.sha }}"/g' env/0x-api/prod/common/07-rfq-cache-worker.yml env/0x-api/prod/v4/*.yml
-            - name: Create Pull Request
-              uses: peter-evans/create-pull-request@v3.5.0
-              with:
-                  token: ${{ secrets.PAT }}
-                  commit-message: '[0xApi][prod] Deploy ${{ steps.parse_pr_info.outputs.ref }}'
-                  title: '[0xApi][prod] Deploy ${{ steps.parse_pr_info.outputs.ref }}'
-                  reviewers: dekz
-                  labels: 0xApi, prod
-                  branch: '0xApi/prod/${{ steps.parse_pr_info.outputs.ref }}'
-                  body: |
-                      https://github.com/${{ github.repository }}/compare/master...${{ steps.parse_pr_info.outputs.ref }}
-                      ${{ steps.parse_pr_info.outputs.ref }}
-            - name: Check outputs
-              run: |
-                  echo "Pull Request Number - ${{ env.PULL_REQUEST_NUMBER }}"
-                  echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
-    manual_prod:
-        runs-on: ubuntu-latest
-        if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/master'
-        steps:
-            - name: Checkout infra repo
+            - name: deploy from master to production
+              if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/master'
+              run: echo "folder=prod" >> $GITHUB_ENV
+            - name: deploy from branch to staging
+              if: github.event.issue.pull_request && github.event.comment.body == 'deploy staging'
+              run: echo "folder=staging" >> $GITHUB_ENV
+            - name: deploy from branch to production
```

Comment:
I kinda want to get rid of this one so that we're forced to deploy from master - anyone else into it?